language/tools/move-cli/src/main.rs

```diff
@@ -842,15 +797,15 @@ fn main() -> Result<()> {
             source_files,
             no_republish,
         } => {
-            move_args.prepare_mode(true)?;
+            move_args.prepare_mode()?;
```

Comment:
It looks like almost every command (except `test` and `clean`) does at least one of the following two things:
- call `prepare_mode` (which creates an `OnDiskStateView`)
- create a subsequent `OnDiskStateView`

In light of this, might make sense to create a single `OnDiskStateView` before the `match` (e.g., it could be returned by `prepare_mode`) and pass it to each of the relevant command functions?

Reply:
It makes sense! I'll make an update.

Reply:
Hmm.... having a second thought now... It seems that `test`, `clean`, and `view` do not involve any dependent libraries so we might not want to create a `storage` directory and splash the library files there.

But despite of this, it still makes sense to have `prepare_mode()` return an `OnDiskStateView` and pass it to other handlers (i.e., `publish`, 'run`, and `doctor`).

Reply:
Nice, that seems like a good middle ground.